telemetry_core/src/main/java/com/newrelic/telemetry/spans/SpanBatchSenderBuilder.java

```diff
@@ -110,6 +114,20 @@ public SpanBatchSenderBuilder traceUrl(URL traceUrl) {
     return this;
   }
 
+  /**
+   * Provide additional user agent information. The product is required to be non-null and
+   * non-empty. The version is optional, although highly recommended.
+   */
+  public SpanBatchSenderBuilder additionalUserAgent(String product, String version) {
+    Utils.verifyNonNull(product, "Product cannot be null in the additional user-agent.");
```

Comment:
What if `product` is an empty string? `verifyNonNull` only checks nullness.

Reply:
good call. I'll add a method to `Utils` to check for null or empty Strings.